membership.md

```diff
@@ -0,0 +1,170 @@
+holistic acceptance of a contribution including: backwards / forwards
+compatibility, adhering to API and flag conventions, subtle performance and
+correctness issues, interactions with other parts of the system, etc.
+
+**Defined by:** *approvers* entry in an OWNERS file in a repo owned by the Kubeflow project.
+
+Approver status is scoped to a part of the codebase.
+
+### Requirements
+
+The following apply to the part of codebase for which one would be an approver in an [OWNERS] file (for repos using the bot).
+
+- Reviewer of the codebase for at least 3 months
+- Primary reviewer for at least 10 substantial PRs to the codebase
+- Reviewed or merged at least 30 PRs to the codebase
+- Nominated by a subproject owner
+  - With no objections from other subproject owners
+  - Done through PR to update the top-level OWNERS file
+
+**Note:** Working Group Leads may nominate and approve `Approvers` that don't meet these requirements due to exceptional circumstances. While acceptable in the short term, Working Group Leads should ensure that these `Approvers` eventually meet the requirements
+
+### Responsibilities and privileges
+
+The following apply to the part of codebase for which one would be an approver
+in an [OWNERS] file (for repos using the bot).
+
+- Approver status may be a precondition to accepting large code contributions
+- Demonstrate sound technical judgement
+- Responsible for project quality control via code reviews
+  - Focus on holistic acceptance of contribution such as dependencies with other features, backwards / forwards
+    compatibility, API and flag definitions, etc
+- Expected to be responsive to review requests
+- Mentor contributors and reviewers
+- May approve code contributions for acceptance
+
+## Inactive members
+
+*Members are continuously active contributors in the community.*
+
+A core principle in maintaining a healthy community is encouraging active
+participation. It is inevitable that people's focuses will change over time and
+they are not expected to be actively contributing forever.
+
+However, being a member of one of the Kubeflow GitHub organizations comes with
+an [elevated set of permissions]. These capabilities should not be used by those
+that are not familiar with the current state of the Kubeflow project.
+
+Therefore members with an extended period away from the project with no activity
+will be removed from the Kubeflow GitHub Organizations and will be required to
+go through the org membership process again after re-familiarizing themselves
+with the current state.
+
+### How inactivity is measured
+
+Inactive members are defined as members of one of the Kubeflow Organizations
+with **no** contributions across any organization within 12 months.
+
+After an extended period away from the project with no activity
+those members would need to re-familiarize themselves with the current state
+before being able to contribute effectively.
```

Comment:
Should we define the policy for updating the OWNER files ? E.g. if contributors don't active on the specific Kubeflow projects.

Reply:
@andreyvelich I commented the same thing above. Is there a way we could plan to automate this process? Maybe CNCF (or other open source projects) have bots that scan contributions automatically every X days and open PRs to clean up owners files. This always needs human verification, in case a contributor is not doing code contributions and the bots cannot detect their work. I imagine the bot opening a PR, adding the steering committee as reviewers

Reply:
following for interest on the automation capabilities offered by CNCF, and +1

Reply:
Kubernetes does have some automation they appear to use. I didn't want to boil the ocean immediately -- We should get some clear criteria in place before following up later with automation.